File: src/MintADB.Wpf/Models/AppPreset.cs
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace MintADB.Wpf.Models;

public sealed class AppPreset : INotifyPropertyChanged
{
    private bool _selected;
    private bool _installed;

    public string Name { get; init; } = "";
    public string Package { get; init; } = "";
    public string[] Processes { get; init; } = [];
    public string[] Services { get; init; } = [];

    public bool Selected
    {
        get => _selected;
        set { if (_selected == value) return; _selected = value; OnPropertyChanged(); }
    }

    public bool Installed
    {
        get => _installed;
        set
        {
            if (_installed == value) return;
            _installed = value;
            OnPropertyChanged();
            OnPropertyChanged(nameof(StatusText));
            OnPropertyChanged(nameof(IsAvailable));
        }
    }

    public bool IsAvailable => _installed;

    public string StatusText => _installed ? "Đã cài" : "Chưa cài";

    public event PropertyChangedEventHandler? PropertyChanged;

    private void OnPropertyChanged([CallerMemberName] string? name = null) =>
        PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));

    public static IReadOnlyList<AppPreset> Defaults { get; } =
    [
        new() { Name = "Telegram", Package = "org.telegram.messenger",
            Processes = ["org.telegram.messenger", "org.telegram.messenger:push"],
            Services = ["org.telegram.messenger/com.google.firebase.messaging.FirebaseMessagingService"] },
        new() { Name = "WhatsApp", Package = "com.whatsapp",
            Processes = ["com.whatsapp", "com.whatsapp:push"],
            Services = ["com.whatsapp/com.google.firebase.messaging.FirebaseMessagingService"] },
        new() { Name = "Gmail", Package = "com.google.android.gm",
            Processes = ["com.google.android.gm", "com.google.android.gm:background"],
            Services = ["com.google.android.gm/com.google.android.gm.provider.MailSyncAdapterService"] },
        new() { Name = "Zalo", Package = "com.zing.zalo",
            Processes = ["com.zing.zalo", "com.zing.zalo:push", "com.zing.zalo:service", "com.zing.zalo:background"],
            Services = [
                "com.zing.zalo/.service.ZaloFirebaseMessagingService",
                "com.zing.zalo/com.google.firebase.messaging.FirebaseMessagingService",
            ] },
        new() { Name = "Messenger", Package = "com.facebook.orca",
            Processes = ["com.facebook.orca", "com.facebook.orca:push", "com.facebook.orca:service"],
            Services = [
                "com.facebook.orca/com.facebook.push.fcm.FcmListenerService",
                "com.facebook.push.fcm.FcmListenerService",
            ] },
        new() { Name = "GMS", Package = "com.google.android.gms",
            Processes = ["com.google.android.gms", "com.google.android.gms:persistent"],
            Services = ["com.google.android.gms/.chimera.PersistentIntentOperationService"] },
        new() { Name = "Play Store", Package = "com.android.vending" },
        new() { Name = "Discord", Package = "com.discord" },
        new() { Name = "Outlook", Package = "com.microsoft.office.outlook" },
        new() { Name = "Slack", Package = "com.Slack" },
        new() { Name = "LINE", Package = "jp.naver.line.android" },
    ];
}

File: src/MintADB.Android/app/src/main/java/com/minthd/mintadb/core/AppPreset.kt
package com.minthd.mintadb.core

data class AppPreset(
    val name: String,
    val packageName: String,
    val processes: List<String> = emptyList(),
    val services: List<String> = emptyList(),
) {
    companion object {
        val defaults = listOf(
            AppPreset("Telegram", "org.telegram.messenger",
                processes = listOf("org.telegram.messenger", "org.telegram.messenger:push"),
                services = listOf("org.telegram.messenger/com.google.firebase.messaging.FirebaseMessagingService")),
            AppPreset("WhatsApp", "com.whatsapp",
                processes = listOf("com.whatsapp", "com.whatsapp:push"),
                services = listOf("com.whatsapp/com.google.firebase.messaging.FirebaseMessagingService")),
            AppPreset("Gmail", "com.google.android.gm",
                processes = listOf("com.google.android.gm", "com.google.android.gm:background")),
            AppPreset("Zalo", "com.zing.zalo",
                processes = listOf("com.zing.zalo", "com.zing.zalo:push", "com.zing.zalo:service")),
            AppPreset("Messenger", "com.facebook.orca",
                processes = listOf("com.facebook.orca", "com.facebook.orca:push")),
            AppPreset("GMS", "com.google.android.gms",
                processes = listOf("com.google.android.gms", "com.google.android.gms:persistent")),
            AppPreset("Play Store", "com.android.vending"),
            AppPreset("Discord", "com.discord"),
            AppPreset("Outlook", "com.microsoft.office.outlook"),
            AppPreset("Slack", "com.Slack"),
            AppPreset("LINE", "jp.naver.line.android"),
        )
    }
}